web/src/SidebarPin.tsx

```diff
@@ -52,11 +52,17 @@ export function SidebarPinContextProvider(
 ) {
   let lsc = useContext(localStorageContext)
 
+  function getInitialCount(): string[] {
+    let ret = lsc.get<Array<string>>("pinned-resources") ?? []
+    incr("ui.web.pin", {
```

Comment:
i'm not sure you can guarantee this will only be called once? i think it get called whenever a new component is created...maybe a global?

Reply:
hm, I was unable to get it to be called multiple times, and the [docs](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state) say:
> If the initial state is the result of an expensive computation, you may provide a function instead, which will be executed only on the initial render

I've switched it to a `useEffect` with zero dependencies, which I have greater confidence won't be re-called. I'd like to verify it with a test, but since I haven't been able to come up with any conditions that would have caused multiple calls in the first place, that'd be tricky!

I'm not inclined to spend too much more time digging into this since even if we can't guarantee there's only one call, the damage is almost certainly minimal.